app/js/script.js

```diff
@@ -0,0 +1,160 @@
+let dragging = false;
+let clockInterval;
+let selectedTimeZone = 'UTC';
+let lastTime = { hours: 0, minutes: 0, seconds: 0 };
+
+const hourHand = document.querySelector('#hour');
+const minuteHand = document.querySelector('#minute');
+const secondHand = document.querySelector('#second');
+
+const timeZones = [
+  'UTC',
+  'America/New_York',
+  'Europe/London',
+  'Asia/Tokyo',
+  'Australia/Sydney'
+];
+
+function startClock() {
+  addTimeZoneSelector();
+  initializeTime();
+
+  [hourHand, minuteHand, secondHand].forEach(hand => {
+    hand.addEventListener('mousedown', startDragging);
+  });
+
+  document.addEventListener('mousemove', handleDragging);
+  document.addEventListener('mouseup', stopDragging);
+
+  clockInterval = setInterval(updateClock, 1000);
+}
+
+function addTimeZoneSelector() {
+  const mainDiv = document.querySelector('.main');
+  const selectorDiv = document.createElement('div');
+
+  const label = document.createElement('label');
+  label.innerText = 'Select Time Zone: ';
+  selectorDiv.appendChild(label);
+
+  const select = document.createElement('select');
+  select.addEventListener('change', (event) => {
+    selectedTimeZone = event.target.value;
+    initializeTime();
+  });
+
+  timeZones.forEach(timeZone => {
+    const option = document.createElement('option');
+    option.value = timeZone;
+    option.innerText = timeZone;
+    select.appendChild(option);
+  });
+
+  selectorDiv.appendChild(select);
+  mainDiv.insertBefore(selectorDiv, mainDiv.firstChild);
+}
+
+function initializeTime() {
+  const currDate = new Date();
+  const utcDate = new Date(currDate.toLocaleString('en-US', { timeZone: 'UTC' }));
+  const timeZoneDate = new Date(utcDate.toLocaleString('en-US', { timeZone: selectedTimeZone }));
+
+  lastTime.seconds = timeZoneDate.getSeconds();
+  lastTime.minutes = timeZoneDate.getMinutes();
+  lastTime.hours = timeZoneDate.getHours() % 12;
+
+  updateClock();
```

Comment:
After you manually set the time, the hands return to the previous position instead of staying where you set them. The InitializeTime function sets lastTime based on the current time of the selected time zone, which overrides the manual adjustments. To fix this, you  probably should update the lastTime object directly based on the positions of the clock hands after you stop dragging.